olad/RDMHTTPModule.cpp

```diff
@@ -3482,6 +3510,203 @@ string RDMHTTPModule::SetCurve(const HTTPRequest *request,
+  section.AddItem(new UIntItem("Minimum Level Upper Limit",
+                               info.min_level_upper_limit));
+  section.AddItem(new UIntItem("Maximum Level Lower Limit",
+                               info.max_level_lower_limit));
+  section.AddItem(new UIntItem("Maximum Level Upper Limit",
+                               info.max_level_upper_limit));
+  section.AddItem(new UIntItem("# of Supported Curves", info.curves_supported));
+  section.AddItem(new UIntItem("Levels Resolution", info.resolution));
+  section.AddItem(new UIntItem("Split Levels Supported",
+                               info.split_levels_supported));
+
+  RespondWithSection(response, section);
+}
+
+
+/**
+ * @brief Handle the request for Dimmer Minimum Levels.
+ */
+string RDMHTTPModule::GetDimmerMinimumLevels(HTTPResponse *response,
+                                             unsigned int universe_id,
+                                             const UID &uid) {
+  string error;
+  m_rdm_api.GetDimmerMinimumLevels(
+    universe_id,
+    uid,
+    ola::rdm::ROOT_RDM_DEVICE,
+    NewSingleCallback(this,
+                      &RDMHTTPModule::GetDimmerMinimumLevelsHandler,
+                      response),
+    &error);
+  return error;
+}
+
+
+/*
+ * @brief Handle the response to a dimmer minimum levels and build the response
+ */
+void RDMHTTPModule::GetDimmerMinimumLevelsHandler(
+    HTTPResponse *response,
+    const ola::rdm::ResponseStatus &status,
+    const ola::rdm::DimmerMinimumDescriptor &info) {
+  if (CheckForRDMError(response, status)) {
+    return;
+  }
+
+  JsonSection section;
+  section.AddItem(new UIntItem("Minimum Level - Increasing",
+                               info.min_level_increasing,
+                               DIMMER_MINIMUM_INCREASING_FIELD));
+  section.AddItem(new UIntItem("Minimum Level - Decreasing",
+                               info.min_level_decreasing,
+                               DIMMER_MINIMUM_DECREASING_FIELD));
+  section.AddItem(new BoolItem("On Below Minimum", info.on_below_min,
+                               GENERIC_BOOL_FIELD));
+
+  RespondWithSection(response, section);
+}
+
+
+/**
+ * @brief Set the dimmer minimum levels
+ */
+string RDMHTTPModule::SetDimmerMinimumLevels(const HTTPRequest *request,
+                                             HTTPResponse *response,
+                                             unsigned int universe_id,
+                                             const UID &uid) {
+  string level = request->GetParameter(DIMMER_MINIMUM_INCREASING_FIELD);
+  uint16_t min_increasing, min_decreasing;
+  uint8_t on_below_min;
+
+  if (!StringToInt(level, &min_increasing)) {
+    return "Invalid minimum level - increasing";
+  }
+
+  level = request->GetParameter(DIMMER_MINIMUM_DECREASING_FIELD);
+
+  if (!StringToInt(level, &min_decreasing)) {
+    return "Invalid minimum level - decreasing";
+  }
+
+  level = request->GetParameter(GENERIC_BOOL_FIELD);
```

Comment:
See SetIdentifyDevice for a more robust way of handling this now it's a bool.

Reply:
I checked the string against empty, but not sure if that is necessary. Seems like it would be able to help give a better error message in the case something goes wrong with the api.

Reply:
Looks good, if you're feeling really keen you could backport that fix to the other places where we use a bool like SetIdentifyDevice. It'll also help for people using the API directly.

A minor nit, but could you either use a second string variable for this, or rename it to be more generic, "level" isn't really appropriate for this, perhaps something like raw_value so it can be reused throughout SetDimmerMinimumLevels?

Reply:
Done.